File: erezervisi_api/eRezervisi/eRezervisi.Common.Dtos/AccommodationUnit/AccommodationUnitUpdateDto.cs
using eRezervisi.Common.Dtos.AccommodationUnitPolicy;
using eRezervisi.Common.Dtos.Image;

namespace eRezervisi.Common.Dtos.AccommodationUnit
{
    public class AccommodationUnitUpdateDto
    {
        public string Title { get; set; } = null!;
        public double Price { get; set; }
        public string? Note { get; set; }
        public string Address { get; set; } = null!;
        public AccommodationUnitPolicyCreateDto Policy { get; set; } = null!;
        public List<ImageUpdateDto> Files { get; set; } = new();
        public long AccommodationUnitCategoryId { get; set; }
        public long TownshipId { get; set; }
        public double Latitude { get; set; }
        public double Longitude { get; set; }
        public double TapPosition { get; set; }
    }
}

File: erezervisi_desktop/erezervisi_desktop/lib/models/requests/accommodation_unit/accommodation_unit_update_dto.dart
import 'package:erezervisi_desktop/models/requests/accommodation_unit/policy_create_dto.dart';
import 'package:erezervisi_desktop/models/requests/image/image_create_dto.dart';

class AccommodationUnitUpdateDto {
  late num id;
  late String title;
  late num price;
  String? note;
  late PolicyCreateDto policy;
  late num categoryId;
  late List<ImageCreateDto> files;
  late num townshipId;
  late num latitude;
  late num longitude;
  late String address;

  AccommodationUnitUpdateDto(
      {required this.id,
      required this.title,
      required this.price,
      this.note,
      required this.policy,
      required this.categoryId,
      required this.files,
      required this.townshipId,
      required this.latitude,
      required this.longitude,
      required this.address});

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'price': price,
      'note': note,
      'policy': policy.toJson(),
      'accommodationUnitCategoryId': categoryId,
      'files': files.map((file) => file.toJson()).toList(),
      'townshipId': townshipId.toString(),
      'latitude': latitude.toString(),
      'longitude': longitude.toString(),
      'address': address
    };
  }
}
